src/Features/CSharp/Portable/Completion/KeywordRecommenders/DynamicKeywordRecommender.cs

```diff
@@ -27,14 +28,11 @@ private static bool IsValidContext(int position, CSharpSyntaxContext context, Ca
             return IsDynamicTypeContext(position, context, cancellationToken);
         }
 
-        public Task<IEnumerable<RecommendedKeyword>> RecommendKeywordsAsync(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
+        public Task<ImmutableArray<RecommendedKeyword>> RecommendKeywordsAsync(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (IsValidContext(position, context, cancellationToken))
-            {
-                return Task.FromResult(SpecializedCollections.SingletonEnumerable(new RecommendedKeyword("dynamic")));
-            }
-
-            return SpecializedTasks.Null<IEnumerable<RecommendedKeyword>>();
+            return IsValidContext(position, context, cancellationToken)
+                ? Task.FromResult(ImmutableArray.Create(new RecommendedKeyword("dynamic")))
```

Comment:
💡 Assuming that `RecommendedKeyword` is immutable, this result could be cached in a `static readonly` field

Reply:
yes.  it *definitely* could.  however, that's much more manual.  something i'd like to do later, but not hugely important right now.